File: archive/scripts/upload_ak_side_dish_images.py
"""
upload_ak_side_dish_images.py

Python alternative to upload_ak_side_dish_images.js
Downloads 10 Ambitious Kitchen side dish images and uploads to Firebase Storage.

Usage:
    pip3 install google-cloud-storage
    python3 upload_ak_side_dish_images.py
"""

import json, os, urllib.request, urllib.error
from pathlib import Path

# ── Firebase / GCS ─────────────────────────────────────────────────────────────
try:
    from google.cloud import storage
    from google.oauth2 import service_account
except ImportError:
    print("Missing dependency. Run:  pip3 install google-cloud-storage")
    raise SystemExit(1)

BUCKET_NAME    = "ckc-recipe-swipe.firebasestorage.app"
SERVICE_ACCT   = Path(__file__).parent / "service-account.json"

creds  = service_account.Credentials.from_service_account_file(str(SERVICE_ACCT))
client = storage.Client(credentials=creds, project=creds.project_id)
bucket = client.bucket(BUCKET_NAME)

# ── Image map: recipe slug → source URL ───────────────────────────────────────
AK_IMAGES = {
    "best-healthy-coleslaw-ever-no-mayo":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2022/07/coleslaw2-5long.jpg",
    "au-gratin-potatoes":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/03/Gratin-FB.png",
    "30-minute-grilled-veggie-orzo":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/08/Brown-Butter-Goat-Cheese-Veggie-Orzo-with-Basil-7long.jpg",
    "lightened-up-cheddar-cauliflower-broccoli-soup":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/Soup-Fb.png",
    "italian-chopped-brussels-sprouts-salad":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/Italian-Chopped-Brussels-Salad-4long.jpg",
    "curry-cashew-chickpea-quinoa-salad":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2020/06/Salad-FB.png",
    "thai-broccoli-salad":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2024/02/Salad-FB.png",
    "curry-roasted-cauliflower-sweet-potato-salad":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2019/01/Curry-Roasted-Sweet-Potato-Cauliflower-Salad-1long.jpg",
    "lightened-sweet-potato-casserole-pecan-oat-streusel":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/11/sweetpotatocasserolelong.jpg",
    "sweet-potato-kale-salad":
        "https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/California-Roasted-Sweet-Potato-Kale-Salad-5long.jpg",
}

# ── Helpers ────────────────────────────────────────────────────────────────────
def ext_and_mime(url):
    base = url.split("?")[0].lower()
    if base.endswith(".png"):
        return ".png", "image/png"
    return ".jpg", "image/jpeg"

def download(url):
    req = urllib.request.Request(url, headers={"User-Agent": "Mozilla/5.0"})
    with urllib.request.urlopen(req, timeout=30) as resp:
        return resp.read()

def upload(slug, data, mime, ext):
    dest = f"images/{slug}{ext}"
    blob = bucket.blob(dest)
    blob.upload_from_string(data, content_type=mime)
    blob.make_public()
    return f"https://storage.googleapis.com/{BUCKET_NAME}/{dest}"

# ── Main ───────────────────────────────────────────────────────────────────────
results, errors = [], []

print(f"Uploading {len(AK_IMAGES)} Ambitious Kitchen side dish images...\n")

for slug, src_url in AK_IMAGES.items():
    print(f"[{slug}]")
    print(f"  Source : {src_url}")
    try:
        ext, mime = ext_and_mime(src_url)
        data = download(src_url)
        print(f"  Downloaded: {len(data)/1024:.1f} KB")
        public_url = upload(slug, data, mime, ext)
        print(f"  Uploaded : images/{slug}{ext}")
        print(f"  URL      : {public_url}")
        results.append({"slug": slug, "publicUrl": public_url})
    except Exception as e:
        print(f"  ERROR: {e}")
        errors.append({"slug": slug, "error": str(e)})
    print()

print("─" * 50)
print(f"Done: {len(results)} uploaded, {len(errors)} failed")

out_path = Path(__file__).parent / "ak_upload_results.json"
with open(out_path, "w") as f:
    json.dump({"uploaded": results, "errors": errors}, f, indent=2)
print(f"Results saved to ak_upload_results.json")

File: upload_ak_side_dish_images.js
/**
 * upload_ak_side_dish_images.js
 *
 * Downloads the 10 Ambitious Kitchen side dish recipe images and uploads
 * them to Firebase Storage under the same path convention as other recipe images.
 *
 * Usage:
 *   node upload_ak_side_dish_images.js
 *
 * Requirements:
 *   npm install node-fetch firebase-admin   (if not already installed)
 *   service-account.json must be present in this directory
 */

const fetch = (...args) => import('node-fetch').then(({default: f}) => f(...args));
const admin = require('firebase-admin');
const path  = require('path');
const fs    = require('fs');

// ── Firebase init ──────────────────────────────────────────────────────────
const serviceAccount = require('./service-account.json');
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  storageBucket: 'ckc-recipe-swipe.firebasestorage.app',
});
const bucket = admin.storage().bucket();

// ── Image map: recipe slug → source URL ───────────────────────────────────
const AK_IMAGES = {
  'best-healthy-coleslaw-ever-no-mayo':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2022/07/coleslaw2-5long.jpg',
  'au-gratin-potatoes':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/03/Gratin-FB.png',
  '30-minute-grilled-veggie-orzo':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/08/Brown-Butter-Goat-Cheese-Veggie-Orzo-with-Basil-7long.jpg',
  'lightened-up-cheddar-cauliflower-broccoli-soup':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/Soup-Fb.png',
  'italian-chopped-brussels-sprouts-salad':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/Italian-Chopped-Brussels-Salad-4long.jpg',
  'curry-cashew-chickpea-quinoa-salad':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2020/06/Salad-FB.png',
  'thai-broccoli-salad':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2024/02/Salad-FB.png',
  'curry-roasted-cauliflower-sweet-potato-salad':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2019/01/Curry-Roasted-Sweet-Potato-Cauliflower-Salad-1long.jpg',
  'lightened-sweet-potato-casserole-pecan-oat-streusel':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/11/sweetpotatocasserolelong.jpg',
  'sweet-potato-kale-salad':
    'https://www.ambitiouskitchen.com/wp-content/uploads/2021/01/California-Roasted-Sweet-Potato-Kale-Salad-5long.jpg',
};

// ── Helpers ────────────────────────────────────────────────────────────────
function getExt(url) {
  const base = url.split('?')[0];
  return path.extname(base).toLowerCase() || '.jpg';
}

function mimeFromExt(ext) {
  return ext === '.png' ? 'image/png' : 'image/jpeg';
}

async function downloadAndUpload(slug, srcUrl) {
  const ext        = getExt(srcUrl);
  const destPath   = `images/${slug}${ext}`;
  const publicUrl  = `https://storage.googleapis.com/ckc-recipe-swipe.firebasestorage.app/${destPath}`;

  console.log(`\n[${slug}]`);
  console.log(`  Source : ${srcUrl}`);

  // Download
  const resp = await fetch(srcUrl, {
    headers: { 'User-Agent': 'Mozilla/5.0 (compatible; recipe-tool/1.0)' }
  });
  if (!resp.ok) throw new Error(`HTTP ${resp.status} from ${srcUrl}`);
  const buffer = Buffer.from(await resp.arrayBuffer());
  console.log(`  Downloaded: ${(buffer.length / 1024).toFixed(1)} KB`);

  // Upload to Firebase Storage
  const file = bucket.file(destPath);
  await file.save(buffer, {
    metadata: { contentType: mimeFromExt(ext) },
    public: true,
  });

  console.log(`  Uploaded : ${destPath}`);
  console.log(`  URL      : ${publicUrl}`);
  return { slug, publicUrl };
}

// ── Main ───────────────────────────────────────────────────────────────────
(async () => {
  console.log(`Uploading ${Object.keys(AK_IMAGES).length} Ambitious Kitchen side dish images...\n`);
  const results = [];
  const errors  = [];

  for (const [slug, srcUrl] of Object.entries(AK_IMAGES)) {
    try {
      const r = await downloadAndUpload(slug, srcUrl);
      results.push(r);
    } catch (err) {
      console.error(`  ERROR: ${err.message}`);
      errors.push({ slug, error: err.message });
    }
  }

  console.log(`\n─────────────────────────────────────────`);
  console.log(`Done: ${results.length} uploaded, ${errors.length} failed`);

  // Write results to a JSON file for reference
  const out = {
    uploaded: results,
    errors,
    firebaseStorageBase: 'https://storage.googleapis.com/ckc-recipe-swipe.firebasestorage.app/images/',
  };
  fs.writeFileSync('ak_upload_results.json', JSON.stringify(out, null, 2));
  console.log(`Results saved to ak_upload_results.json`);
})();
